src/main/java/bio/terra/service/snapshot/flight/create/CreateSnapshotParquetFilesAzureInterface.java

```diff
@@ -19,59 +17,43 @@
 import java.util.UUID;
 import java.util.stream.Collectors;
 
-public class CreateSnapshotParquetFilesAzureStep implements Step {
+public interface CreateSnapshotParquetFilesAzureInterface {
+  default StepResult createSnapshotParquetFiles(
+      FlightContext context, AzureSynapsePdao azureSynapsePdao, SnapshotService snapshotService)
+      throws InterruptedException {
+    try {
+      Map<String, Long> tableRowCounts = createSnapshotPrimaryDataParquetFiles(context);
+      createRowIdsAndStoreRowCount(context, azureSynapsePdao, snapshotService, tableRowCounts);
```

Comment:
Idea: There are really two main steps within this step. We really could pull this into two separate steps and the createRowIdsAndStoreRowCount would be shared between all azure snapshots. And then we could consider getting rid of this interface.

Reply:
I like this idea! Having more granular steps that do fewer things better aligns with Stairway recommendations.